lua/loaders/from_vscode.lua

```diff
@@ -41,23 +41,26 @@ local function file_read(path)
 end
 
 
-local function load_snippet_file(lang, snippet_set_path)
+local function load_snippet_file(langs, snippet_set_path)
     if not path_exists(snippet_set_path) then return end
-    local lang_snips = ls.snippets[lang] or {}
-
     local snippet_set_data = json_decode(file_read(snippet_set_path))
-    for name, parts in pairs(snippet_set_data) do
-        local body = type(parts.body) == "string" and parts.body or table.concat(parts.body, '\n')
-
-        -- There are still some snippets that fail while loading
-        pcall(function()
-        table.insert(
-            lang_snips,
-            ls.parser.parse_snippet({trig=parts.prefix, name=name, wordTrig=true}, body)
-        )
-        end) 
+
+    for _, lang in pairs(langs) do
+        local lang_snips = ls.snippets[lang] or {}
+
+        for name, parts in pairs(snippet_set_data) do
+            local body = type(parts.body) == "string" and parts.body or table.concat(parts.body, '\n')
+
+            -- There are still some snippets that fail while loading
+            pcall(function()
+            table.insert(
+                lang_snips,
+                ls.parser.parse_snippet({trig=parts.prefix, name=name, wordTrig=true}, body)
+            )
+            end) 
+        end
```

Comment:
First of all, thanks :D
Second, are you certain that wrapping `table.insert` into a function is necessary? Am I missing something? :sweat_smile:

Reply:
Yes, the parser can fail if the snippet is wrong.

Reply:
Yeah, I agree with the `pcall`, I just think `function() table.insert(...) end` can be simplified into `table.insert(...)`.

Reply:
pcall needs a function as parameter

Reply:
Ah I see now, my mistake, thought pcall also error-checks the args but it doesn't.